receiver/receiverhelper/obsreport.go

```diff
@@ -104,14 +114,16 @@ func (rec *ObsReport) EndLogsOp(
 	numReceivedLogRecords int,
 	err error,
 ) {
+	rec.recordDuration(receiverCtx)
 	rec.endOp(receiverCtx, format, numReceivedLogRecords, err, pipeline.SignalLogs)
 }
 
 // StartMetricsOp is called when a request is received from a client.
 // The returned context should be used in other calls to the obsreport functions
 // dealing with the same receive operation.
 func (rec *ObsReport) StartMetricsOp(operationCtx context.Context) context.Context {
-	return rec.startOp(operationCtx, internal.ReceiverMetricsOperationSuffix)
+	ctx := rec.startOp(operationCtx, internal.ReceiverMetricsOperationSuffix)
+	return context.WithValue(ctx, ctxKeyReceiverStartTime{}, time.Now())
```

Comment:
This is tested indirectly, as we run the code.
But we don't assert that the metric is indeed recorded (or that it isn't if `EndMetricsOp` runs without the context value.
Could you add those tests?